tiledb/sm/query/readers/ordered_dim_label_reader.cc

```diff
@@ -0,0 +1,914 @@
+    per_range_array_tile_indexes[r].resize(fragment_metadata_.size());
+  }
+  throw_if_not_ok(parallel_for_2d(
+      storage_manager_->compute_tp(),
+      0,
+      fragment_metadata_.size(),
+      0,
+      ranges_.size(),
+      [&](uint64_t f, uint64_t r) {
+        per_range_array_tile_indexes[r][f] =
+            get_array_tile_indexes_for_range(f, r);
+        return Status::Ok();
+      }));
+
+  // Compute the tile indexes (min/max) that can potentially contain the label
+  // value for each range start/end.
+  per_range_array_tile_indexes_.resize(ranges_.size());
+  throw_if_not_ok(parallel_for(
+      storage_manager_->compute_tp(), 0, ranges_.size(), [&](uint64_t r) {
+        per_range_array_tile_indexes_[r] = RangeTileIndexes(
+            tile_idx_min_, tile_idx_max_, per_range_array_tile_indexes[r]);
+        return Status::Ok();
+      }));
+}
+
+void OrderedDimLabelReader::load_label_min_max_values() {
+  auto timer_se = stats_->start_timer("load_label_min_max_values");
+  const auto encryption_key = array_->encryption_key();
+
+  // Load min/max data for all fragments.
+  throw_if_not_ok(parallel_for(
+      storage_manager_->compute_tp(),
+      0,
+      fragment_metadata_.size(),
+      [&](const uint64_t i) {
+        auto& fragment = fragment_metadata_[i];
+        std::vector<std::string> names_min = {label_name_};
+        RETURN_NOT_OK(fragment->load_tile_min_values(
+            *encryption_key, std::move(names_min)));
+        std::vector<std::string> names_max = {label_name_};
+        RETURN_NOT_OK(fragment->load_tile_max_values(
+            *encryption_key, std::move(names_max)));
+        return Status::Ok();
+      }));
+}
+
+template <typename LabelType>
+OrderedDimLabelReader::FragmentRangeTileIndexes
+OrderedDimLabelReader::get_array_tile_indexes_for_range(
+    unsigned f, uint64_t r) {
+  const auto tile_num = fragment_metadata_[f]->tile_num();
+  uint64_t start_index = 0, end_index = tile_num - 1;
+
+  const auto start_range = get_range_as<LabelType>(r, 0);
+  const auto end_range = get_range_as<LabelType>(r, 1);
+
+  // Check if either the start or end range is fully excluded from the fragment.
+  IndexValueType start_val_type = IndexValueType::CONTAINED,
+                 end_val_type = IndexValueType::CONTAINED;
+
+  if (increasing_labels_) {
+    // Start with the minimum of the first tile.
+    const auto min =
+        fragment_metadata_[f]->get_tile_min_as<LabelType>(label_name_, 0);
+    if (start_range < min) {
+      start_val_type = IndexValueType::LT;
+    }
+    if (end_range < min) {
+      end_val_type = IndexValueType::LT;
+    }
+
+    // Now with the maximum of the last tile.
+    const auto max = fragment_metadata_[f]->get_tile_max_as<LabelType>(
+        label_name_, tile_num - 1);
+    if (start_range > max) {
+      start_val_type = IndexValueType::GT;
+    }
+    if (end_range > max) {
+      end_val_type = IndexValueType::GT;
+    }
```

Comment:
I think it would be a little easier to read if we combine if statements. Something like the following:

```suggestion
    const auto min =
        fragment_metadata_[f]->get_tile_min_as<LabelType>(label_name_, 0);
    const auto max = fragment_metadata_[f]->get_tile_max_as<LabelType>(
        label_name_, tile_num - 1);
        
     if (start_range < min) {
      start_val_type = IndexValueType::LT;
     else if (start_range > max) {
      start_val_type = IndexValueType::GT;
     }
     
    if (end_range < min) {
      end_val_type = IndexValueType::LT;
    } else if (end_range > max) {
      end_val_type = IndexValueType::GT;
    }
```